### DISCUSSION TITLE
HeaderGroups with rowspan

### DISCUSSION THREAD

**Discussion by aspirisen** [TanStack/table#4359] [Q&A]
Hello, i use headers with nested columns, and looks like react-table does not calculate appropriate rowSpan for the header cells.

<img width="1465" alt="image" src="https://user-images.githubusercontent.com/3620639/188270725-0ab0ae72-d8be-415b-8000-8668ea5b499b.png">
None-nested columns do not have rowspan to look like single-level column header.

Also react-table inserts cells, so i can't manually calculate rowSpan as there are real cells.
However i would expect react-table to automatically calculate correct rowSpan as looks like there is enough information about header group size

I have columns like that, as you can see i have nested columns
```
[
    {
        "id": "1",
        "Header": "One"
    },
    {
        "id": "2",
        "Header": "Two",
        "columns": [
            {
                "id": "3",
                "Header": "Three"
            },
            {
                "id": "4",
                "Header": "Four"
            }
        ]
    },
    {
        "id": "5",
        "Header": "Five"
    },
    {
        "id": "6",
        "Header": "Second"
    },
    {
        "id": "7",
        "Header": "Seven"
    },
    {
        "id": "8",
        "Header": "Eight"
    },
    {
        "id": "9",
        "Header": "Nine"
    },
    {
        "id": "10",
        "Header": "Ten",
        "columns": [
            {
                "id": "11",
                "Header": "Eleven"
            },
            {
                "id": "12",
                "Header": "Twelve"
            }
        ]
    }
]
```

There is similar discussion, but not about automatic header group rowspan calculation https://github.com/TanStack/table/discussions/2233

**Comment by Darcrandex** [comment]
I have the same question, there is a [demo](https://codesandbox.io/p/sandbox/jolly-tree-nm7vvs)

**Comment by Roman86** [comment]
Just solved this problem and packaged it into a tiny npm library. It's really simple to use — you can rowspan both the header and footer, with table and non-table tr/th elements. Check it out!:

```
npm i tanstack-table-header-rowspan
```

https://www.npmjs.com/package/tanstack-table-header-rowspan
